src/sass/modules/_m-schemaform.scss

```diff
@@ -77,3 +81,36 @@
 .schemaform-array-row {
   outline: none;
 }
+
+.schemaform-block-header {
+  margin-bottom: 0.5em;
+  > legend, > h5, > p {
+    padding-bottom: 0 !important;
+    margin-bottom: 0 !important;
+  }
+  > p {
+    margin-top: 0.5em;
+  }
+}
+
+// This is the worst
```

Comment:
I updated these styles to (mostly) use specific classes, which hopefully makes it clearer. These are at least really specific and unlikely to cause problems later, aside from missing more deeply nested fields, which we can fix as we hit them.